x-pack/plugins/security_solution/common/api/detection_engine/signals_migration/finalize_signals_migration/finalize_signals_migration.gen.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { z } from 'zod';
+
+/*
+ * NOTICE: Do not edit this file manually.
+ * This file is automatically generated by the OpenAPI Generator, @kbn/openapi-generator.
+ *
+ * info:
+ *   title: Finalize alerts migration API endpoint
+ *   version: 2023-10-31
+ */
+
+export type FinalizeAlertsMigrationRequestBody = z.infer<typeof FinalizeAlertsMigrationRequestBody>;
+export const FinalizeAlertsMigrationRequestBody = z.object({
+  migration_ids: z.array(z.string()).min(1).optional(),
+});
+
+export type FinalizeAlertsMigrationResponse = z.infer<typeof FinalizeAlertsMigrationResponse>;
+export const FinalizeAlertsMigrationResponse = z.array(MigrationFinalizationResult);
+
+export type MigrationFinalizationResult = z.infer<typeof MigrationFinalizationResult>;
+export const MigrationFinalizationResult = z.object({
+  id: z.string(),
+  completed: z.boolean(),
+  destinationIndex: z.string(),
+  status: z.enum(['success', 'failure', 'pending']),
+  sourceIndex: z.string(),
+  version: z.string(),
+  updated: z.string().datetime(),
+  error: z
+    .object({
+      message: z.string().optional(),
+      status_code: z.number().int().optional(),
```

Comment:
`message` and `status_code` within error are not optional, though error is optional